Src/PerformanceCollector/Filtering/DocumentStream.cs

```diff
@@ -0,0 +1,239 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.Helpers;
+
+    /// <summary>
+    /// Represents a concept of a uniquely identifiable set of full telemetry documents that are being reported by the SDK. 
```

Comment:
There should be configuration per stream? Why Configuration has list of streams and not the other way around? Why this class also has lists per type. Are those lists configurations or actual values